src/Microsoft.ML.TensorFlow/TensorflowCatalog.cs

```diff
@@ -100,9 +102,42 @@ public static class TensorflowCatalog
         /// <param name="catalog">The transform's catalog.</param>
         /// <param name="options">The <see cref="TensorFlowEstimator.Options"/> specifying the inputs and the settings of the <see cref="TensorFlowEstimator"/>.</param>
         /// <param name="tensorFlowModel">The pre-loaded TensorFlow model.</param>
-        public static TensorFlowEstimator TensorFlow(this TransformsCatalog catalog,
+        public static TensorFlowEstimator TensorFlow(this TransformsCatalog.TensorFlowTransforms catalog,
             TensorFlowEstimator.Options options,
             TensorFlowModelInfo tensorFlowModel)
             => new TensorFlowEstimator(CatalogUtils.GetEnvironment(catalog), options, tensorFlowModel);
+
+        /// <summary>
+        /// This method retrieves the information about the graph nodes of a TensorFlow model as an <see cref="DataViewSchema"/>.
+        /// For every node in the graph that has an output type that is compatible with the types supported by
+        /// <see cref="TensorFlowEstimator"/>, the output schema contains a column with the name of that node, and the
+        /// type of its output (including the item type and the shape, if it is known). Every column also contains metadata
+        /// of kind <see cref="TensorFlowUtils.TensorflowOperatorTypeKind"/>, indicating the operation type of the node, and if that node has inputs in the graph,
+        /// it contains metadata of kind <see cref="TensorFlowUtils.TensorflowUpstreamOperatorsKind"/>, indicating the names of the input nodes.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="modelLocation">Location of the TensorFlow model.</param>
+        public static DataViewSchema GetModelSchema(this TransformsCatalog.TensorFlowTransforms catalog, string modelLocation)
+            => TensorFlowUtils.GetModelSchema(CatalogUtils.GetEnvironment(catalog), modelLocation);
+
+        /// <summary>
+        /// This is a convenience method for iterating over the nodes of a TensorFlow model graph. It
+        /// iterates over the columns of the <see cref="DataViewSchema"/> returned by <see cref="GetModelSchema(TransformsCatalog.TensorFlowTransforms, string)"/>,
+        /// and for each one it returns a tuple containing the name, operation type, column type and an array of input node names.
+        /// This method is convenient for filtering nodes based on certain criteria, for example, by the operation type.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="modelLocation">Location of the TensorFlow model.</param>
+        public static IEnumerable<(string, string, DataViewType, string[])> GetModelNodes(this TransformsCatalog.TensorFlowTransforms catalog, string modelLocation)
```

Comment:
>GetModelNodes [](start = 76, length = 13)

This doesn't need to be a part of the public API.

Reply:
It is being use here.
[https://github.com/dotnet/machinelearning/blob/fb6ce5442f1af7b30dc48f6a438b2a7b63779390/src/Microsoft.ML.DnnAnalyzer/Microsoft.ML.DnnAnalyzer/DnnAnalyzer.cs#L20](https://github.com/dotnet/machinelearning/blob/fb6ce5442f1af7b30dc48f6a438b2a7b63779390/src/Microsoft.ML.DnnAnalyzer/Microsoft.ML.DnnAnalyzer/DnnAnalyzer.cs#L20)

If I don't expose it like this then I will have to make the internal one public.

---
In reply to: [258728605](https://github.com/dotnet/machinelearning/pull/2672#discussion_r258728605) [](ancestors = 258728605)